store/book-api.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { Module, VuexModule,Mutation, Action } from 'vuex-module-decorators'
+
+import axios from 'axios'
+
+import { saveAuthSession, clearAuthSession, loadAuthSession} from '~/utils/auth'
+import { API_POST_AUTHORIZE } from '~/constant/api'
+
+@Module({
+  name: 'book-api',
+  stateFactory: true,
+  namespaced: true,
+})
+export default class BookAPI extends VuexModule {
+  token = ''
+  sessionWallet = ''
+  isRestoringSession = false
+
+  @Mutation
+  setToken(inputToken: string) {
+    this.token = inputToken
+  }
+
+  @Mutation
+  setSessionWallet(inputWallet: string) {
+    this.sessionWallet = inputWallet
+  }
+
+  @Mutation
+  setIsRestoringSession(inputIsRestoringSession: boolean) {
+    this.isRestoringSession = inputIsRestoringSession
+  }
+
+  @Action
+  async authenticate({inputWallet = '', signature = {}}: {inputWallet?: string, signature?: any}) {
+    try {
+      const { data } = await axios.post(
+        API_POST_AUTHORIZE,
+        signature,
+      )
+      const token = (data as any)?.token
+      if (!token) {
+        throw new Error('INVALID_SIGNATURE')
+      }
+      this.context.commit('setToken', token)
+      this.context.commit('setSessionWallet', inputWallet)
+      saveAuthSession({ wallet: inputWallet, token})
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
+      throw new Error('AUTHENTICATION_FAILED')
```

Comment:
Maybe include the http body in the error message throw, if the error is an axios response

Reply:
[🥅 Throw error with axios response data](https://github.com/likecoin/app-like-co/pull/474/commits/11d36aca5a66a60714fe5a71bd340513cb7fad3b)